src/audio/codec_adapter/codec/cadence.c

```diff
@@ -171,7 +173,22 @@ static int apply_config(struct comp_dev *dev, enum codec_cfg_type type)
 		param = data;
 		comp_dbg(dev, "apply_config() applying param %d value %d",
 			 param->id, param->data[0]);
-		/* Set read parameter */
+		/* Update stream parameters */
+		switch (param->id) {
+		case CADENCE_PARAM_SAMPLE_RATE:
+			param->data[0] = stream->rate;
+			break;
+		case CADENCE_PARAM_PCM_WDSZ:
+			param->data[0] = (stream->frame_fmt == SOF_IPC_FRAME_S16_LE) ?
+				       0x10 : 0x20; /* 16 or 32 bit samples */
+			break;
+		case CADENCE_PARAM_NUM_CHANNELS:
+			param->data[0] = stream->channels;
+			break;
+		default:
+			break;
+		}
```

Comment:
Good point, pity codec which apply to the same API, Cadence in this case define parameters differently :|

Reply:
how about encoding these 3 parameters in the topology, I mean the IDs? We can assign them during prepare and use accordingly. I don't see a better solution for now..

Reply:
Good point - can we make this really clear in the comments as no doubt this will catch a lot of folks out !!!

Reply:
@mrajwa yes, encoding the IDs in the topology exactly how they were supposed to be defined by the Cadence is a good idea. Then we need to to an internal translation so that we can override the exact parameters that we want (sample rate, etc).